src/krati_Jain/Shop.java

```diff
@@ -0,0 +1,114 @@
+package krati_Jain;
+
+public class Shop {
+
+	int initialMaggieQty;
+	int initialDosaQty;
+	int initialOilQty;
+	int initialPanipuriQty;  
+	int initialMasalaQty;
+
+void setInitialQty(int maggie, int dosa, int oil, int panipuri, int masala) {
+	initialMaggieQty = maggie;
+	initialDosaQty = dosa;
+	initialOilQty = oil;
+	initialPanipuriQty = panipuri;  
+	initialMasalaQty = masala;
+}
+
+void customerPurchase(int maggieQty, int dosaQty, int oilQty, int panipuriQty, int masalaQty) {
+	if (maggieQty > 0 ) {
+		if (maggieQty <= initialMaggieQty) {
+			System.out.println("Maggie packets sold to the customer are: " + maggieQty);
+			initialMaggieQty = initialMaggieQty - maggieQty;
+			} else {
+			System.out.println("Maggie packets sold to the customer are: " + initialMaggieQty);
+			initialMaggieQty = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct